clippy_lints/src/cognitive_complexity.rs

```diff
@@ -20,7 +22,9 @@ declare_clippy_lint! {
     /// **Known problems:** Sometimes it's hard to find a way to reduce the
     /// complexity.
     ///
-    /// **Example:** No. You'll see it when you get the warning.
+    /// **Example:** Sorry. Examples are too big and varied to put in here. For a
+    /// complete explanation of the analysis being made though, you can read this paper:
+    /// https://www.sonarsource.com/docs/CognitiveComplexity.pdf
```

Comment:
You may want to link to a README somewhere that has a fuller list of examples; the .pdf will not be so Rust specific and it seems there will be plenty of Rust specific aspects to this lint.

Reply:
Ohh, that's true! I guess a README is in order :)